src/features/Account/ManageAccountForm.tsx

```diff
@@ -0,0 +1,319 @@
+  const history = useHistory();
+
+  // Is the form submission currently being processed? (loading state)
+  const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
+
+  // Track the details of the account that is either being created or updated
+  const [accountDetails, setAccountDetails] = useState<IAccount>({
+    accountType: (getValueFromQuery('accountType') as UserType) || UserType.UNKNOWN,
+    birthDate: '',
+    confirmed: false,
+    email: getNestedAttr(props, ['location', 'state', 'email']) || '',
+    firstName: '',
+    id: '',
+    lastName: '',
+    password: getNestedAttr(props, ['location', 'state', 'password']) || '',
+    phoneNumber: '',
+    pronoun: '',
+    gender: '',
+    dietaryRestrictions: [],
+  })
+
+  // If this is an edit form, load current account's data as default value
+  // for account details
+  useEffect(() => {
+    (async () => {
+      if (props.mode === ManageAccountModes.EDIT) {
+        try {
+          const response = await Account.getSelf();
+          const newAccountDetails = response.data.data;
+          newAccountDetails.birthDate = date2input(newAccountDetails.birthDate);
+          setAccountDetails(newAccountDetails);
+        } catch (e) {
+          // If can't find self's account, shouldn't be logged in. Redirect to home page
+          history.push(FrontendRoute.HOME_PAGE)
+        }
+      }
+    })();
+  }, []);
+
+  /**
+   * This converts a formik values object into the IAccount object.
+   * @param values Formik values
+   * @param accountId the account id associated with this hacker.
+   */
+  const convertFormikToAccount = (values: FormikValues, accountId: string = ''): IAccount => (
+    {
+      accountType: UserType.UNKNOWN,
+      birthDate: input2date(values.birthDate),
+      confirmed: false,
+      email: values.email,
+      firstName: values.firstName,
+      id: accountId,
+      lastName: values.lastName,
+      password: values.password,
+      phoneNumber: values.phoneNumber,
+      pronoun: values.pronoun,
+      gender: values.gender,
+      dietaryRestrictions: values.dietaryRestrictions,
+    }
+  );
+
+  // Handle form submission
+  const handleSubmit = (values: FormikValues) => {
+    // Record that form is being submitted
+    setIsSubmitting(true);
+
+    // Turn data entered into form fields into an IAccount
+    const formattedDetails = convertFormikToAccount(values, accountDetails.id);
+
+    // Depending on the mode of this form either create a new account with form data
+    // or update user's existing account to match form data
+    if (props.mode === ManageAccountModes.CREATE) {
+      handleCreate(formattedDetails);
+    } else if (props.mode === ManageAccountModes.EDIT) {
+      handleEdit(formattedDetails, values.password, values.newPassword);
+    }
+
+    // Once submitted, redirect user to appropriate page
+    if (isSponsor(accountDetails)) {
+      history.push(FrontendRoute.CREATE_SPONSOR_PAGE);
```

Comment:
Can you test this behavior with a sponsor account that is editing their profile? What does this do?

Reply:
I missed this case when redoing the redirect code, this should be `if (props.mode === ManageAccountModes.CREATE && isSponsor(accountDetails))`

Reply:
Feel free to resolve after you implement the mentioned change & then test.